contracts/validator-manager/PoSValidatorManager.sol

```diff
@@ -562,30 +515,32 @@ abstract contract PoSValidatorManager is
         PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
         Delegator memory delegator = $._delegatorStakes[delegationID];
 
-        // Unpack the Warp message
-        WarpMessage memory warpMessage = _getPChainWarpMessage(messageIndex);
-        (bytes32 validationID, uint64 nonce,) =
-            ValidatorMessages.unpackSubnetValidatorWeightUpdateMessage(warpMessage.payload);
-
-        if (delegator.validationID != validationID) {
-            revert InvalidValidationID();
-        }
-
-        // The received nonce should be at least as high as the delegation's ending nonce. This allows a weight
-        // update using a higher nonce (which implicitly includes the delegation's weight update) to be used to
-        // complete delisting for an earlier delegation. This is necessary because the P-Chain is only willing
-        // to sign the latest weight update.
-        if (delegator.endingNonce > nonce) {
-            revert InvalidNonce();
-        }
-
         // Ensure the delegator is pending removed. Since anybody can call this function once
         // end delegation has been initialized, we need to make sure that this function is only
         // callable after that has been done.
         if (delegator.status != DelegatorStatus.PendingRemoved) {
```

Comment:
In a case where the validator ends before the delegator, is `initializeEndDelegation` intended to be the only function allowed to be called to complete the delegation? If so, can we make that clear in some of the documentation/comments?

Reply:
We could leave an escape in `completeEndDelegation` if the validator is `Active` and the validator is `Completed`, but I fear that would make the code a little circular.